# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContext.h
# Commit: 4cff94ca9818
# Full Hash: 4cff94ca98189769f21a855520a345c6c4e5c841
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2021-01-15 03:50:53
# Regressor Bug: 1640607
# File Overlap Count: 1
# Description:
#   Bug 1640607 - Send SurfaceDescriptors for GPU blitting for video-to-webgl. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101061
# ==============================================================================

diff -r 044f8b594c71 -r 4cff94ca9818 dom/canvas/WebGLContext.h
--- a/dom/canvas/WebGLContext.h	Thu Jan 14 01:20:12 2021 +0000
+++ b/dom/canvas/WebGLContext.h	Thu Jan 14 01:23:06 2021 +0000
@@ -1237,35 +1237,7 @@
 
   template <typename... Args>
   void GeneratePerfWarning(const char* const fmt, const Args&... args) const {
-    if (!ShouldGeneratePerfWarnings()) return;
-
-    const auto funcName = FuncName();
-    nsCString msg;
-    msg.AppendPrintf("WebGL perf warning: %s: ", funcName);
-
-#ifdef __clang__
-#  pragma clang diagnostic push
-#  pragma clang diagnostic ignored "-Wformat-security"
-#elif defined(__GNUC__)
-#  pragma GCC diagnostic push
-#  pragma GCC diagnostic ignored "-Wformat-security"
-#endif
-    msg.AppendPrintf(fmt, args...);
-#ifdef __clang__
-#  pragma clang diagnostic pop
-#elif defined(__GNUC__)
-#  pragma GCC diagnostic pop
-#endif
-
-    GenerateErrorImpl(0, msg);
-
-    mNumPerfWarnings++;
-    if (!ShouldGeneratePerfWarnings()) {
-      GenerateWarning(
-          "After reporting %u, no further WebGL perf warnings will"
-          " be reported for this WebGL context.",
-          uint32_t(mNumPerfWarnings));
-    }
+    GenerateError(webgl::kErrorPerfWarning, fmt, args...);
   }
 
  public: